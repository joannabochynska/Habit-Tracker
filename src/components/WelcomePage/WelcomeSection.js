import React from 'react';
import PropTypes from 'prop-types';
import styles from './WelcomeSection.module.scss';

const renderImage = (image, header) => (
    <div className={styles.imageSection}>
        <img src={ image } alt={ header }/>
    </div>
);

const renderDetails = (header, text) => (
    <div className={styles.details}>
        <h4>{ header }</h4>
        <p>{ text }</p>
    </div>
);

const WelcomeSectionComponent = ({ header, text, image, textPlacement }) => (
    <div className={styles.section}>
        { textPlacement === "left"
            ? <div className={ `${styles.row} ${styles.left}`}>
                { renderDetails(header, text) }
                { renderImage(image, header) }
            </div>
            : <div className={ `${styles.row} ${styles.right}`}>
                { renderImage(image, header) }
                { renderDetails(header, text) }
            </div>
        }
    </div>
);

WelcomeSectionComponent.propTypes = {
    header: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    textPlacement: PropTypes.oneOf(['left', 'right']).isRequired,
}

export default WelcomeSectionComponent;