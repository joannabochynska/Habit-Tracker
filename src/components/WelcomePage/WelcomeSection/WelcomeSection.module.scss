@import "../../../scss/constants";

.section {
    align-items: center;
    display: grid;
    grid-template-columns: 5% auto 5%;
    padding-top: 80px;
    padding-bottom: 80px;

    @media (max-width: $mobileBreakpoint) {
        padding-top: 20px;
        padding-bottom: 20px;
    }
}

.row {
    grid-column: 2;
    display: grid;
    grid-template-columns: 45% 10% 45%;
    align-items: center;


    @media (max-width: $mobileBreakpoint) {
        display: flex;
        flex-direction: column;
    }

    .details {
        align-self: start;

        h4 {
            color: $headersTextColor;
            font-size: 56px;
            font-family: $mainFont;
            font-weight: 400;
            margin-bottom: 0;

            @media (max-width: $mobileBreakpoint) {
                font-size: 28px;
                margin-top: 20px;
            }
        }

        p {
            margin: 30px 0;
            color: $textColor;
            font-family: $secondFont;
            font-size: 20px;
            line-height: 1.5;

            @media (max-width: $mobileBreakpoint) {
                font-size: 16px;
            }
        }
    }

    .imageSection {
        img {
            width: 100%;
        }
    }
}


.left {
    .details {
        grid-column: 1;
    }

    .imageSection {
        grid-column: 3;
    }
}

.right {
    .details {
        grid-column: 3;
    }

    .imageSection {
        grid-column: 1;
    }
}
