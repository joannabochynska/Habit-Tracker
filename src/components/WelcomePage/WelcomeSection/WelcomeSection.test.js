import React from 'react';
import renderer from 'react-test-renderer';
import WelcomeSection from './WelcomeSection';

test('renders correctly', () => {
  const tree = renderer
    .create(<WelcomeSection
      header="test header"
      text="test text"
      image="testImageUrl"
      textPlacement="left"
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
