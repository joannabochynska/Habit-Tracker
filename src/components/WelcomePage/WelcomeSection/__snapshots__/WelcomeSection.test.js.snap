// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="section"
>
  <div
    className="row left"
  >
    <div
      className="details"
    >
      <h4>
        test header
      </h4>
      <p>
        test text
      </p>
    </div>
    <div
      className="imageSection"
    >
      <img
        alt="test header"
        src="testImageUrl"
      />
    </div>
  </div>
</div>
`;
